packages/lib-core/src/store/PluginLoader.ts

```diff
@@ -304,10 +343,146 @@ export class PluginLoader {
+
+    this.plugins.set(pluginName, pluginLoadData);
+    const initPromise = new Promise<void>((resolve, reject) => {
+      const script = getDocument().createElement('script');
+
+      script.src = resolveURL(baseURL, pluginLoadData.manifest.entryScript);
+      script.async = true;
+
+      script.onload = async (event) => {
+        if (onLoad) {
+          onLoad(event);
+        }
+        if (pluginLoadData.manifest.libraryType === 'var') {
+          await this.initializeVarModule(pluginName);
+        }
+        return resolve();
+      };
+
+      script.onerror = (event) => {
+        if (onError) {
+          onError(event);
+        }
+        return reject(new Error('Unable to initialize remote module'));
+      };
+
+      getDocument().head.appendChild(script);
+
+      consoleLogger.info(`Loading entry script of plugin ${pluginName} from ${script.src}`);
+    });
+
+    return initPromise;
+  }
+
+  /**
+   * Initialize container loaded to window scope.
+   *
+   * Required by HCC plugins.
+   * Simulates the jsonp flow used in the other type of plugins.
+   */
+  async initializeVarModule(pluginName: string) {
+    if (!this.plugins.has(pluginName)) {
+      throw new ErrorWithCause(`No plugin data exist for '${pluginName}'`);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const data = this.plugins.get(pluginName)!;
+    const moduleScope = data.manifest.name;
+    if (moduleScope) {
+      await initSharedScope(this.options.sharedScopeName);
+      const entryModule: PluginEntryModule = (
+        window as unknown as { [key: string]: PluginEntryModule }
+      )[moduleScope];
+      try {
+        await Promise.resolve(entryModule.init(this.options.sharedScope));
+      } catch (error) {
+        data.status = 'failed';
+        this.invokeListeners({
+          success: false,
+          pluginName,
+          errorCause: 'Plugin container initialization failed',
+          errorMessage: `Unable to initialize ${pluginName} webpack container! ${error}`,
+        });
+        return;
+      }
+      data.status = 'loaded';
+      data.entryCallbackFired = true;
+      this.invokeListeners({
+        success: true,
+        pluginName,
+        manifest: data.manifest,
+        entryModule,
+      });
+    } else {
+      data.status = 'failed';
+      this.invokeListeners({
+        success: false,
+        pluginName,
+        errorCause: 'Ivalid var container manifest.',
+        errorMessage: `The attribute 'name' is missing in '${pluginName}' plugin manifest!`,
+      });
+    }
+  }
+
+  setPluginData(data: PluginLoadData) {
+    this.plugins.set(data.manifest.name, data);
+  }
+
+  static isUniqueScript(
+    scriptURL: string,
+    getDocument: () => typeof document = _.constant(document),
+  ) {
+    return getDocument().head.querySelectorAll(`script[src="${scriptURL}"]`).length === 0;
+  }
+
+  static async loadPluginRuntimeScript(
```

Comment:
I still wonder if this added complexity is really worth it (supporting a separate webpack runtime chunk) but I guess we need it for some plugins.

We should update the sample plugin to emit a separate runtime chunk so this code can be easily exercised.

Martin, in case of plugin reload - not reloading the runtime chunk script is OK, because its content doesn't change across plugin builds, is that correct?

Reply:
The runtime chunk will not change. It only provides the ability for additional webpack chunk loading. I was doing some testing where a plugin has changed in the sample app and it worked well.

I think we are at a point where adding some e2e tests is a good idea.